src/lib.rs

```diff
@@ -1698,8 +1830,9 @@ impl Stream {
             )
         };
         if Status::failed(status) {
-            panic!("StreamSend failure 0x{:x}", status);
+            return Err(status);
         }
+        Ok(())
```

Comment:
1. Does Rust allow for non-zero `Ok` status codes? For instance, we have the 'pending' status.
2. Should we return `Ok(status)` here?